File: java/spring-arch-unit/spring-arch-unit-mvc-example/src/test/java/com/bmarques/springarchunitmvcexample/ArchUnitPackageCheckTest.java
package com.bmarques.springarchunitmvcexample;

import com.tngtech.archunit.core.domain.JavaClasses;
import com.tngtech.archunit.core.importer.ClassFileImporter;
import com.tngtech.archunit.lang.ArchRule;
import org.junit.jupiter.api.Test;

import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;

public class ArchUnitPackageCheckTest {

    @Test
    public void invoiceClassesOnlyShouldResideInInvoicePackage() {
        JavaClasses importedClasses = new ClassFileImporter().importPackages("com.bmarques.springarchunitmvcexample.domain");

        ArchRule myRule = classes().that().haveSimpleNameStartingWith("Invoice")
                .should().resideInAPackage("com.bmarques.springarchunitmvcexample.domain.invoice");

        myRule.check(importedClasses);
    }


    @Test
    public void invoiceClassesOnlyShouldResideInInvoicePackageUsingPrefixInvoice() {
        JavaClasses importedClasses = new ClassFileImporter().importPackages("com.bmarques.springarchunitmvcexample.domain");

        ArchRule myRule = classes().that().haveSimpleNameContaining("Invoice")
                .should().resideInAPackage("com.bmarques.springarchunitmvcexample.domain.invoice");

        myRule.check(importedClasses);
    }
}

File: kotlin/spring-arch-unit/spring-arch-unit-mvc-example/src/test/kotlin/com/bmarques/springarchunitmvcexample/ArchUnitPackageCheckTest.kt
package com.bmarques.springarchunitmvcexample

import com.tngtech.archunit.core.importer.ClassFileImporter
import com.tngtech.archunit.lang.ArchRule
import com.tngtech.archunit.lang.syntax.ArchRuleDefinition
import org.junit.jupiter.api.Test

class ArchUnitPackageCheckTest {

    @Test
    fun `Invoice classes only should reside in invoice package`() {
        val importedClasses = ClassFileImporter().importPackages("com.bmarques.springarchunitmvcexample.domain")
        val myRule: ArchRule = ArchRuleDefinition.classes().that().haveSimpleNameStartingWith("Invoice")
                .should().resideInAPackage("com.bmarques.springarchunitmvcexample.domain.invoice")
        myRule.check(importedClasses)
    }


    @Test
    fun `Invoice classes only should reside in invoice package using prefix invoice`() {
        val importedClasses = ClassFileImporter().importPackages("com.bmarques.springarchunitmvcexample.domain")
        val myRule: ArchRule = ArchRuleDefinition.classes().that().haveSimpleNameContaining("Invoice")
                .should().resideInAPackage("com.bmarques.springarchunitmvcexample.domain.invoice")
        myRule.check(importedClasses)
    }
}